libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -32,6 +32,21 @@ void AP_Camera_Backend::update()
     // check feedback pin
     check_feedback();
 
+    // time based triggering
+    // if time and distance trigerring both are enabled then we only do time based triggering
+    if (time_interval_settings.enabled) {
+        uint32_t interval_ms = AP_HAL::millis() - last_picture_time_ms;
+        if (time_interval_settings.num_remaining && (interval_ms >time_interval_settings.time_interval*1000.0f)) {
+            if (take_picture()) {
```

Comment:
If we can't take a picture for some reason we will just keep on trying to take pictures at 50hz which is quite fast.  I guess this is OK though... I guess this won't cause any harm.